securedrop/crypto_util.py

```diff
@@ -172,8 +174,16 @@ def genrandomid(self,
                         for x in range(words_in_random_id))
 
     def display_id(self):
-        return ' '.join([random.choice(self.adjectives),
-                         random.choice(self.nouns)])
+        """Generate random journalist_designation until we get an unused one"""
```

Comment:
Is there a security issue with changing this from a constant-time operation to a loop that might leak information about the number of sources in the database? (given the likely size of SD databases, his is more theoretical, but it might be better to have the source generation just fail altogether and prompt the user to restart the process.)

Reply:
@zenmonkeykstop Not sure how this would reveal the number of sources in the database.

Reply:
It shouldn't take too many iterations to avoid a collision, so probably wouldn't give a good idea of the number of sources. As @emkll pointed out, the unique index on `journalist_designation` would help decrease the time this check takes, too, but as [discussed in the issue](https://github.com/freedomofpress/securedrop/issues/2043#issuecomment-658991658), that's going to be a more involved undertaking which we'll tackle later.

It would be good in any case to have an upper bound on the number of attempts. I think 50 would be quick enough while allowing for some really bad luck. If that limit is hit, we can flash an error and ask the visitor to retry.

Reply:
@rmol Sounds good. I will add an upper limit to it. Should it be done in the same PR?

Reply:
Yes, it's best to do this as part of the same PR Prateek. Thanks again for working on this. :)

Reply:
@prateekj117 Just touching base, do you still have time to poke at this / would you like help to move it over the finish line?

Reply:
Hey @eloquence, so sorry, had quite some things to wrap up this week. I will complete it within this week for sure.

Reply:
@eloquence What exception should I raise when `display_id` hits duplicate `journalist_designation` 50 times.

Reply:
Will just raise a `ValueError` for now.

Reply:
@prateekj117 Simply raising `ValueError` will work. If you have time, this is a case where I think friendlier error handling than we're consistently providing in the source interface would help not scare away a new source, e.g. in `source_app.main_create`: 

```diff
diff --git i/securedrop/source_app/main.py w/securedrop/source_app/main.py
index 172849fbc..fa5f67e73 100644
--- i/securedrop/source_app/main.py
+++ w/securedrop/source_app/main.py
@@ -73,7 +73,19 @@ def make_blueprint(config):
 
             filesystem_id = current_app.crypto_util.hash_codename(codename)
 
-            source = Source(filesystem_id, current_app.crypto_util.display_id())
+            try:
+                source = Source(filesystem_id, current_app.crypto_util.display_id())
+            except ValueError as e:
+                current_app.logger.error(e)
+                flash(
+                    gettext(
+                        "There was a temporary problem creating your account. "
+                        "Please try again."
+                    ),
+                    "error"
+                )
+                return redirect(url_for('.index'))
+
             db.session.add(source)
             try:
                 db.session.commit()
```

But again, I think it's unlikely that the limit will be hit, and we could stand to make some general improvements here, so this could also be deferred.

Reply:
@rmol Yes, I agree. Something like this would be much better. Will use this only.